docs/book/src/basics/control_flow.md

```diff
@@ -32,7 +32,20 @@ Note that all branches of the `if` expression must return a value of the same ty
 
 ### `match` expressions
 
-Sway supports advanced pattern matching through exhaustive `match` expressions.
+Sway supports advanced pattern matching through exhaustive `match` expressions. Unlike an `if` statement, a match statement asserts at compile time all possible patterns have been matched. If you don't handle all cases, you'll get a warning from the compiler indicating that you need to add a `_`, a catch-all pattern that matches any value and is used when you want to specify a default action.
+
+The basic syntax of a `match` statement:
+
+```sway
+match expression {
```

Comment:
Just to emphasize that it is possible to return a value from a `match` expression.
```suggestion
let result = match expression {
```